### PR TITLE
Fix #1321 - MorphiaLazyModel: Refactor in 14.0.0

### LINKED ISSUE #1321: MorphiaLazyModel: Refactor in 14.0.0
Based on core PF changes: https://github.com/primefaces/primefaces/pull/10776

cc @Rapster cc @newk5

### PR DESCRIPTION
**Pull request by Rapster**
@newk5 for review

### CODE REVIEW

**Comment by newk5** [comment]
Thanks! Sorry I haven't been able to look into this yet, I'm not familiar with the new LazyDatamodel refactor that was done recently in PF. 

I was now looking through this commit and I agree with all changes except one in particular with the builder. I think we should rename the entityManager(etc..) to datastore(etc..) because morphia has no concept of entity managers, entity managers usually mean JPA related stuff. I understand this probably came from the JPALazyDataModel in PF, and wanting to normalize the API to be the same between the JPALazyModel and MorphiaLazyDataModel but for morphia users, the term "entityManager" will probably cause confusion, since morphia always refers to it as "datastore"

**Comment by Rapster** [comment]
Yup, just an oversight

File: core/src/main/java/org/primefaces/extensions/model/mongo/MorphiaLazyDataModel.java [hunk 1]
```diff
@@ -344,41 +325,93 @@ protected Method getRowKeyGetter() {
...
+        public Builder() {
+            model = new MorphiaLazyDataModel<>();
+        }
+
+        public Builder<T> entityClass(Class<T> entityClass) {
+            model.entityClass = entityClass;
+            return this;
+        }
+
+        public Builder<T> entityManager(SerializableSupplier<Datastore> datastore) {
```
Comment (melloware) [inline core/src/main/java/org/primefaces/extensions/model/mongo/MorphiaLazyDataModel.java]: EntityManager to Datastore and then I think this is great!

**Review by melloware — CHANGES_REQUESTED** [review]

**Review by melloware — APPROVED** [review]

**Comment by Rapster** [comment]
Be aware I couldn't test it (or we have a branch for that?)

**Comment by melloware** [comment]
We don't. This code was donated by @newk5 as he uses it heavily in his company.

**Pull request merged**